src/super_gradients/training/processing/processing.py

```diff
@@ -357,6 +371,160 @@ def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
         return cropped_image, None
 
 
+@register_processing(Processings.SegResizeWithPadding)
+class SegResizeWithPadding(Processing):
+    """Resize image to given image dimensions while preserving aspect ratio (padding might be used).
+
+    :param output_shape:    (H, W)
+    :param pad_value:    padding value (will be used if padding needed)
+    """
+
+    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
+        self.output_shape = output_shape
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, SegmentationRescaleWithPaddingMetadata]:
+        height, width = image.shape[:2]
+        scale_factor = min(self.output_shape[0] / height, self.output_shape[1] / width)
+
+        if scale_factor != 1.0:
+            new_height, new_width = round(height * scale_factor), round(width * scale_factor)
+            image = _rescale_image(image, target_shape=(new_height, new_width))
+
+        padding_coordinates = _get_center_padding_coordinates(input_shape=image.shape, output_shape=self.output_shape)
+        processed_image = _pad_image(image=image, padding_coordinates=padding_coordinates, pad_value=self.pad_value)
+
+        return processed_image, SegmentationRescaleWithPaddingMetadata(
+            original_shape=(height, width), scale_factor=scale_factor, padding_coordinates=padding_coordinates
+        )
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: SegmentationRescaleWithPaddingMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = predictions.segmentation_map[
+            metadata.padding_coordinates.top : predictions.segmentation_map_shape[0] - metadata.padding_coordinates.bottom,
+            metadata.padding_coordinates.left : predictions.segmentation_map_shape[1] - metadata.padding_coordinates.right,
+        ]
+        predictions.segmentation_map = _rescale_image(predictions.segmentation_map.astype(np.uint8), target_shape=metadata.original_shape)
+        predictions.segmentation_map = predictions.segmentation_map.astype(np.int64)
+        return predictions
+
+
+@register_processing(Processings.SegmentationRescale)
+class SegmentationRescale(Processing):
+    """Rescale image by scaling factor while preserving aspect ratio.
+    The rescaling can be done according to scale_factor, short_size or long_size.
+    If more than one argument is given, the rescaling mode is determined by this order: scale_factor, then short_size,
+    then long_size.
+
+    :param scale_factor: Rescaling is done by multiplying input size by scale_factor:
+        out_size = (scale_factor * w, scale_factor * h)
+    :param short_size:  Rescaling is done by determining the scale factor by the ratio short_size / min(h, w).
+    :param long_size:   Rescaling is done by determining the scale factor by the ratio long_size / max(h, w).
+    """
+
+    def __init__(self, scale_factor: float, short_size: int, long_size: int):
```

Comment:
I think it is a error prone design. IMO we should not confuse user by providing him a class, which behavior varies that significantly based on arguments.

To me it is much clearer to have two classes with a clear behavior and one parameter:
- SegmentationResizeLongestSizeTo(1024) - Would resize input image such that it's longest size is 1024
- SegmentationResizeShortestSizeTo(1024) - Similarly - would resize input image such that it's shortest size is 1024

Why two transforms and not three - because I cannot imagine any use case where i'd be willing to specify explicit scale factor over the absolute pixel values for resizing.